### DISCUSSION TITLE
Is there support for evasion attacks for regression problems?

### DISCUSSION THREAD

**Discussion by GillianGrayson** [Trusted-AI/adversarial-robustness-toolbox#2160] [Q&A]
Hi!

Thanks for a very cool and useful toolkit!
I have a question: does this package support any evasion attacks for regression problems?
I saw that there is support for 4 types of regression estimators (Black-Box, Keras, PyTorch, Scikit-learn), but all possible evasion attacks have the following initialization in the code:
```python
classifier: "CLASSIFIER_TYPE"
```
where `CLASSIFIER_TYPE` are all possible classifier types listed in the file [art/utils.py](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/utils.py#L170).

There is no evasion attack that supports [REGRESSOR_TYPE](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/utils.py#L200).

Is it even possible to use evasion attacks for regression problems?

**Comment by beat-buesser** [comment]
Hi @GillianGrayson Thank you for you compliments! I think all cases of black-box evasion attacks on regression I have heard of have translated the regression task into a binary classification problem using `BlackBoxClassifier` to define a range within the regression value should change to represent a successful evasion attack on the regression. I think `ProjectedGradientDescentNumpy` should support all regression estimators as a white-box attack maximizing the regression loss.

**Reply by GillianGrayson** [reply]
Thank you so much for your prompt reply!

1. Could you please clarify the details of the transition from the regression problem to binary classification? What will be the class labels? The range of values of the predicted variable will divide by the median into 2 classes and now we will not regress the target variable, but classify its range?
Or this new classification should divide all samples into "bad" and "good" samples in terms of the error (e.g., mean_absolute_error) of the original regression?  It is not quite clear how to train such a model, since it depends on the original regression model.
Maybe you have an example or paper that illustrates this approach?

2. Thank you! `ProjectedGradientDescentNumpy` works fine with [PyTorchRegressor](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/estimators/regression/pytorch.py#L50). Interestingly, `ProjectedGradientDescentPyTorch` did NOT work with [PyTorchRegressor](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/estimators/regression/pytorch.py#L50), because, unlike `ProjectedGradientDescentNumpy` it is additionally derived from [<class 'art.estimators.classification.classifier.ClassifierMixin'>](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/attacks/evasion/projected_gradient_descent/projected_gradient_descent_pytorch.py#L60), which makes it applicable to classification problems only.
